content/cookbook/tokens/create-nft.md

```diff
@@ -0,0 +1,179 @@
+---
+title: How to create an NFT
+sidebarSortOrder: 15
+description: "Learn how to create an NFT on Solana, using Arweave and Metaplex."
+---
+
+To create an NFT you have to:
+
+1. Upload the image to IPFS like Arweave
+2. Upload the json metadata to IPFS like Arweave
+3. Call metaplex to create an account for the NFT
+
+### Upload to Arweave
+
+```typescript filename="upload-to-arweave.ts"
+import fs from "node:fs";
+import Arweave from "arweave";
+
+(async () => {
+  const arweave = Arweave.init({
+    host: "localhost",
+    port: 1984,
+    protocol: "http",
+    timeout: 20000,
+    logging: false,
+  });
+
+  const host = arweave.getConfig().api.host;
+  const port = arweave.getConfig().api.port;
+  const protocol = arweave.getConfig().api.protocol;
+
+  // Upload image to Arweave
+  const data = fs.readFileSync("./code/nfts/upload-arweave/lowres-dog.png");
+
+  const transaction = await arweave.createTransaction({
+    data: data,
+  });
+
+  transaction.addTag("Content-Type", "image/png");
+
+  // const wallet = JSON.parse(fs.readFileSync("./code/nfts/upload-arweave/wallet.json", "utf-8"))
+  const wallet = await arweave.wallets.generate();
+  const addr = await arweave.wallets.getAddress(wallet);
+  console.log("address ", addr);
+
+  await arweave.api.get(`/mint/${encodeURI(addr)}/10000000000000000`);
+  await arweave.transactions.sign(transaction, wallet);
+
+  const response = await arweave.transactions.post(transaction);
+  console.log(response);
+
+  const id = transaction.id;
+  const imageUrl = id ? `${protocol}://${host}:${port}/${id}` : undefined;
+  console.log("imageUrl", imageUrl);
+
+  // Upload metadata to Arweave
+
+  const metadata = {
+    name: "Custom NFT #1",
+    symbol: "CNFT",
+    description: "A description about my custom NFT #1",
+    seller_fee_basis_points: 500,
+    external_url: "https://www.customnft.com/",
+    attributes: [
+      {
+        trait_type: "NFT type",
+        value: "Custom",
+      },
+    ],
+    collection: {
+      name: "Test Collection",
+      family: "Custom NFTs",
+    },
+    properties: {
+      files: [
+        {
+          uri: imageUrl,
+          type: "image/png",
+        },
+      ],
+      category: "image",
+      maxSupply: 0,
+      creators: [
+        {
+          address: "CBBUMHRmbVUck99mTCip5sHP16kzGj3QTYB8K3XxwmQx",
+          share: 100,
+        },
+      ],
+    },
+    image: imageUrl,
+  };
+
+  const metadataRequest = JSON.stringify(metadata);
+
+  const metadataTransaction = await arweave.createTransaction({
+    data: metadataRequest,
+  });
+
+  metadataTransaction.addTag("Content-Type", "application/json");
+
+  await arweave.transactions.sign(metadataTransaction, wallet);
+
+  console.log("metadata txid", metadataTransaction.id);
+
+  console.log(await arweave.transactions.post(metadataTransaction));
```

Comment:
For some reason, I find `console.log(await`) to be odd. There's nothing technically wrong with it. It just looks strange. What do you think?